Zend/zend_alloc.c

```diff
@@ -2376,6 +2475,18 @@ void zend_mm_shutdown(zend_mm_heap *heap, bool full, bool silent)
 		memset(p->free_map, 0, sizeof(p->free_map) + sizeof(p->map));
 		p->free_map[0] = (1L << ZEND_MM_FIRST_PAGE) - 1;
 		p->map[0] = ZEND_MM_LRUN(ZEND_MM_FIRST_PAGE);
+
+		pid_t pid = getpid();
+		if (heap->pid != pid) {
+			if (zend_mm_init_key(heap) != SUCCESS) {
```

Comment:
No need for a full reinit here: You can `php_random_xoshiro256starstar_jump` (https://www.php.net/manual/en/random-engine-xoshiro256starstar.jump.php). For this use case, it will probably not be more predictable than a full reinit, but it will be much faster, because you don't need the CSPRNG.

Reply:
In forking SAPIs the heap is initialized in a root process and re-used in child processes. If we don't re-seed here, we end up with the same random sequence in all child processes. It's not an issue performance-wise as it's done only once per process.